src/Polysemy/Fish.hs

```diff
@@ -0,0 +1,282 @@
+-- combination with ('@>'). Use ('>@>') instead if you only want to work with
+-- 'Members' of your union.
+--
+-- __Examples:__
+--
+-- List of multiple members:
+--
+-- @
+-- foo :: 'Members' \'['Polysemy.State.State' 'Int', 'Polysemy.Input.Input' 'String'] r => 'Sem' (ConsoleIO:r) () -> 'Sem' r ()
+-- @
+--
+-- can be written as:
+--
+-- @
+-- foo :: ConsoleIO:r \@> () -> \'['Polysemy.State.State' 'Int', 'Polysemy.Input.Input' 'String'] >\@ r \@> ()
+-- @
+--
+-- One member:
+--
+-- @
+-- foo :: 'Member' ('Polysemy.State.State' 'Int') r => 'Sem' (ConsoleIO:r) () -> 'Sem' r ()
+-- @
+--
+-- can be written as both:
+--
+-- @
+-- foo :: ConsoleIO:r \@> () -> \'['Polysemy.State.State' 'Int'] >\@ r \@> ()
+-- @
+--
+-- and:
+--
+-- @
+-- foo :: ConsoleIO:r \@> () -> 'Polysemy.State.State' 'Int' >\@ r \@> ()
+-- @
+--
+-- where effect without list gets put into one automatically.
+--
+-- __Exactly__ one, lifted monad as a member:
+--
+-- @
+-- foo :: 'Member' ('Lift' 'IO') r => 'Sem' (ConsoleIO:r) () -> 'Sem' r ()
+-- @
+--
+-- can be written simply as:
+--
+-- @
+-- foo :: ConsoleIO:r \@> () -> 'IO' >\@ r \@> ()
+-- @
+--
+-- and will be automatically lifted and put into list.
+--
+-- __NOTE:__ For this operator to work, you have to provide types of monad and
+-- contained value in your effect:
+--
+-- @
+-- data MyEffect x y (m :: * -> *) (a :: *) where ...
+-- @
+--
+-- but this requirement may be removed in the future.
+infix 1 >@
+type es >@ s = Members (ToEffList "(>@)" es) (SemList s) => s
+
+-- | Joined version of ('>@') and ('@>') with implicit, hidden list of effects
+-- in union --- suited for actions that only use one 'Sem' in their type.
+--
+-- E.g.:
+--
+-- @
+-- foo :: 'Members' \'['Polysemy.State.State' 'String', 'Polysemy.Input.Input' 'Int'] r => 'String' -> 'Int' -> 'Sem' r ()
+-- @
+--
+-- can be written as:
+--
+-- @
+-- foo :: 'String' -> 'Int' -> \'['Polysemy.State.State' 'String', 'Polysemy.Input.Input' 'Int'] >\@> ()
+-- @
+--
+-- For more examples on how to specify member effects, see ('>@').
+--
+-- __NOTE:__ For this operator to work, you have to provide types of monad and
+-- contained value in your effect:
+--
+-- @
+-- data MyEffect x y (m :: * -> *) (a :: *) where ...
+-- @
+--
+-- but this requirement may be removed in the future.
+infix 1 >@>
+type es >@> a = forall r. Members (ToEffList "(>@>)" es) r => Sem r a
+
+
+------------------------------------------------------------------------------
+-- Miscellaneous -------------------------------------------------------------
+------------------------------------------------------------------------------
+-- | Constraint version of ('>@').
+--
+-- __NOTE:__ For this constraint to work, you have to provide types of monad
+-- and contained value in your effect:
+--
+-- @
+-- data MyEffect x y (m :: * -> *) (a :: *) where ...
+-- @
+--
+-- but this requirement may be removed in the future.
+type Means es r = Members (ToEffList "Means" es) r
```

Comment:
I'm not sure it's worth exposing this --- what do you think is the benefit?

Reply:
Not really, I just added it for consistency. Will be removed.